flink-core/src/main/java/org/apache/flink/api/common/eventtime/CombinedWatermarkStatus.java

```diff
@@ -95,10 +97,12 @@ public boolean updateCombinedWatermark() {
     static class PartialWatermark {
         private long watermark = Long.MIN_VALUE;
         private boolean idle = false;
+
+        @Nullable
```

Comment:
If this is only used for the sake of tests, I would actually prefer to leave this field as non-null with no-op lambda functions.

Reply:
Agreed.